incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/OltpExporterHandler.java

```diff
@@ -0,0 +1,175 @@
+        List<AttributeConfig> attributes)
+    {
+        this.context = context;
+        this.metricsUrl = createMetricsUrl(exporter.options().endpoint);
+        this.signals = exporter.options().signals.signals;
+        this.protocol = exporter.options().endpoint.protocol;
+        this.interval = Duration.ofSeconds(exporter.options().interval).toMillis();
+        this.collector = collector;
+        this.resolveKind = resolveKind;
+        this.attributes = attributes;
+        this.httpClient = HttpClient.newBuilder().build();
+    }
+
+    @Override
+    public void start()
+    {
+        assert signals.contains(METRICS);
+        assert HTTP.equals(protocol);
+
+        MetricsReader metrics = new MetricsReader(collector, context::supplyLocalName);
+        serializer = new OtlpMetricsSerializer(metrics.records(), attributes, context::resolveMetric, resolveKind);
+        lastAttempt = 0;
+        lastSuccess = System.currentTimeMillis();
+        warning = false;
+    }
+
+    @Override
+    public int export()
+    {
+        long now = System.currentTimeMillis();
+        if (now - lastSuccess < interval)
+        {
+            return 0;
+        }
+        if (now - lastAttempt < RETRY_INTERVAL)
+        {
+            return 0;
+        }
+        if (now - lastSuccess >= WARNING_INTERVAL && !warning)
+        {
+            System.out.format("Warning: Could not successfully publish data to OpenTelemetry Collector for %d seconds.%n",
+                Duration.ofMillis(now - lastSuccess).toSeconds());
+            warning = true;
+        }
+        post();
+        return 1;
+    }
+
+    // required for testing
+    URI metricsUrl()
+    {
+        return metricsUrl;
+    }
+
+    private URI createMetricsUrl(
+        OtlpEndpointConfig endpoint)
+    {
+        URI result;
+        URI location = URI.create(endpoint.location);
+        if (endpoint.overrides != null && endpoint.overrides.metrics != null)
+        {
+            URI metricsOverride = URI.create(endpoint.overrides.metrics);
+            if (metricsOverride.isAbsolute())
+            {
+                result = metricsOverride;
+            }
+            else
+            {
+                result = location.resolve(metricsOverride);
+            }
+        }
+        else
+        {
+            result = location.resolve(DEFAULT_METRICS_URI);
+        }
+        return result;
+    }
+
+    private void post()
+    {
+        String json = serializer.serializeAll();
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(metricsUrl)
+            .header("Content-Type", "application/json")
+            .POST(HttpRequest.BodyPublishers.ofString(json))
+            .build();
+        lastAttempt = System.currentTimeMillis();
+        CompletableFuture<HttpResponse<String>> response = httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString());
+        response.thenAccept(this::handleResponse);
```

Comment:
This method body can be inlined at the call site, and `lastAttempt` can use `now` instead too.

Reply:
fixed, pls chk